Update workflows.md (#18)
jspm/jspm.org: pages/docs/workflows.md

prev-section = "index"
+++

> # Generating Import Maps

Reviewer: This title is the title of the entire page, are you suggesting renaming this entire section to `https://jspm.org/generating-import-maps` instead of `https://jspm.org/workflow-guides`? Alternatively, we could make this a sub-section.

> The [Online Generator](#online-generator), [VSCode Extension](#vscode-extension), or the [JS Generator API](#js-generator-api) can be used to generate import maps for your project, with the Generator API being the recommended and most direct approach. There is also support for more advanced and experimental [Deno import maps](#deno-import-maps).

Reviewer: Perhaps the sentence should start with the goal before the links?

```suggestion
Import maps can be generated for projects using the [Online Generator](#online-generator), [VSCode Extension](#vscode-extension), or the [JS Generator API](#js-generator-api), with the Generator API being the recommended and most direct approach. There is also support for more advanced and experimental [Deno import maps](#deno-import-maps).
```

Also typically we try to use passive voice - without using words like _you_ or _we_.

The full Node.js resolver rules are supported so it's even possible to use [own-name resolution](https://nodejs.org/dist/latest-v17.x/docs/api/packages.html#self-referencing-a-package-using-its-name) and [package imports resolution](https://nodejs.org/dist/latest-v17.x/docs/api/packages.html#subpath-imports) while respecting custom local import map mappings.

# Development Workflows

> Import maps can be used in development workflows for Browser, Node and Deno environments. Browser application workflows are not required to have a bundling step, and all environments may still employ file watching and hot-reloading. Equally, build steps involving transpilation and using task runners are supported when using import maps. Simply incorporate your preferred method for generating import maps with your preferred workflow.

Reviewer: ```suggestion
Import maps can be used in development workflows for Browser, Node.js and Deno environments. Browser application workflows are not required to have a bundling step, and all environments may still employ file watching and hot-reloading. Equally, build steps involving transpilation and using task runners are supported when using import maps. Simply incorporate your preferred method for generating import maps with your preferred workflow.
```

It's a shame we don't have hot-reloading workflows ready to demo... well we do, but they are still an unreleased Chomp feature.
